src/Common/SettingConstraintType.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+
+namespace DB
+{
+
+enum class SettingConstraintType
```

Comment:
I'd prefer a clearer name, like `SettingConstraintWritability` (I know it's longer).
Because this enum describes only the writability aspect and not some common type.